{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "237e1dfa-99fc-4f51-898f-a967351f64d4",
   "metadata": {},
   "source": [
    "# MayMySections (https://alleninstitute.github.io/MapMySections/)\n",
    "\n",
    "Accurately defining this connection between genetic tools and known cell types represents a critical step in interpreting the results of experiments using these tools, from functional assays to potential gene therapies. A detailed cell type characterization of the labeled cells can be achieved through a combination of single cell RNA-sequencing and cell sorting; however, such methods take time, are costly, and are prone to bias. A method for directly inferring cell types from fluorescent images without the need for additional experiments would be immediately applicable to thousands of existing genetic tools, greatly improving their utility and interpretability. Entrants are tasked with 1) creating an algorithm that will accurately match fluorescent images of genetic tools to the most likely spatial transcriptomic cell types and/or 2) presenting such an algorithm as part of a user-friendly tool like MapMyCells. Although images span the whole brain, this challenge is focused on defining cell types in primary visual cortex (VISp).\n",
    "\n",
    "This challenge includes images and associated cell type specificity for anonymized genetics tools from the Genetics Tools Atlas, a searchable web resource representing information and data on enhancer-adeno-associated viruses (enhancer AAVs) and mouse transgenes. Multiple modalities for summarizing data are included as part of the atlas, but only coronal sections collected using Serial Two-Photon Tomography (STPT) sections will be directly included as part of the challenge. Cell type specificity is assessed by applying single cell RNA-sequencing (SMART-Seq v4) on fluorescently labeled cells from each genetic tool, and then mapping these cells to the published taxonomy of cell types in whole mouse brain (Yao et al, 2023), which are available in the Allen Brain Cell Atlas. Qualitative assessments of cell labeling patterns are also provided for many of the genetic tools."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "62154772-5e05-4ec4-947e-fbb247122686",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "11b40085-afa2-4c10-a1dc-dad1122ead36",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['README', 'Entrant Information', 'Column Descriptions', 'Training Set', 'Test Set']\n"
     ]
    }
   ],
   "source": [
    "workbook = pd.ExcelFile(\"MapMySections_EntrantData.xlsx\")\n",
    "sheets = workbook.sheet_names\n",
    "print(sheets)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "734ad0da-5268-4e38-824e-f6a08e0ef4ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MapMySectionsID : A unique, anonymized identifier for each experiment. These identifiers will be linked to actual experiment identifiers in the Genetics Tools Atlas at the conclusion of the challenge \n",
      "\n",
      "Genetics Tools Type : Either “Enhancer AAV” or “Mouse transgenic line” to indicating the underlying type of genetic tool. Note that the training and test data sets each include some enhancer AAVs and some mouse transgenic lines. \n",
      "\n",
      "STPT Data File Path : Link to the image series in OME-Zarr format. This is the primary data required for the challenge. See Data Challenge page for resources showing how to access and use these data. \n",
      "\n",
      "STPT Thumbnail Image : Link to a small thumbnail image of one section of the genetic tool that intersects VISp for quick viewing. These are potentially useful as a sanity check of algorithm results or for inclusion in user-created tools. \n",
      "\n",
      "Neuroglancer File Path : Link to same data for visualization using Neuroglancer. See Data Challenge page for resources showing how to use Neuroglancer. \n",
      "\n",
      "CCF Registered Image File Path : Link to zip file containing quantified intensities of red and green channels for each 25um CCF voxel in the STPT image series.  See Data Challenge page for resources showing how to access and use these data. \n",
      "\n",
      "Target_Cell_Population : Cell populations targeted with this experiment. Note: This is not necessarily the cell populations actually identified! \n",
      "\n",
      "Qualitative Image Assessment : Sets of comma-separated qualitative assessments for labeling strength, labeling density, and labeled cell populations made based on review of epifluorescence image data for the same genetic tool. In cases where more than one assessment is included, cells from all of the listed types were labeled. While these calls do not constitute quantitative validation, they are likely to reflect actual cell type assignments in cases when SMART-Seq v4 data is missing. Note that a value of \"Neuron\" indicates that the specific neuronal populations targeted could not be accurately assessed in that experiment. \n",
      "\n",
      "(Quantitative Cell Type Assignments) : The remaining columns reflect quantitative cell type information from up to 48 cells from a given genetic tool collected using SMART-Seq v4 and aligned to whole mouse brain cell types from the Allen Brain Atlas at the \"subclass\" level of the hierarchy. These columns represent the ground truth data to be targeted by computational algorithms.  Numbers in these columns represent the count of cells from a given experiment that mapped to the corresponding subclass. \n",
      "\n",
      "ABC.NN : Number of ABC.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Astro.TE.NN : Number of Astro.TE.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "CLA.EPd.CTX.Car3.Glut : Number of CLA.EPd.CTX.Car3.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Endo.NN : Number of Endo.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L2.3.IT.CTX.Glut : Number of L2.3.IT.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L4.5.IT.CTX.Glut : Number of L4.5.IT.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L5.ET.CTX.Glut : Number of L5.ET.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L5.IT.CTX.Glut : Number of L5.IT.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L5.NP.CTX.Glut : Number of L5.NP.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L6.CT.CTX.Glut : Number of L6.CT.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L6.IT.CTX.Glut : Number of L6.IT.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "L6b.CTX.Glut : Number of L6b.CTX.Glut cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Lamp5.Gaba : Number of Lamp5.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Lamp5.Lhx6.Gaba : Number of Lamp5.Lhx6.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Lymphoid.NN : Number of Lymphoid.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Microglia.NN : Number of Microglia.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "OPC.NN : Number of OPC.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Oligo.NN : Number of Oligo.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Peri.NN : Number of Peri.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Pvalb.Gaba : Number of Pvalb.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Pvalb.chandelier.Gaba : Number of Pvalb.chandelier.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "SMC.NN : Number of SMC.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Sncg.Gaba : Number of Sncg.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Sst.Chodl.Gaba : Number of Sst.Chodl.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Sst.Gaba : Number of Sst.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "VLMC.NN : Number of VLMC.NN cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n",
      "Vip.Gaba : Number of Vip.Gaba cells found in SSv4 validation. Blank if no SSv4 validation performed. \n",
      "\n"
     ]
    }
   ],
   "source": [
    "df = workbook.parse(sheets[2])\n",
    "for index, row in df.iterrows():\n",
    "    print(row[\"Column Name\"], \":\", row[\"Description\"], \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "07d49f5c-1bf8-488e-880a-2546d9b63c6e",
   "metadata": {},
   "source": [
    "## Important sections of the dataframe and dataset\n",
    "\n",
    "### Qualitative Image Assessment\n",
    "\n",
    "Sets of comma-separated qualitative assessments for labeling strength, labeling density, and labeled cell populations made based on review of epifluorescence image data for the same genetic tool. In cases where more than one assessment is included, cells from all of the listed types were labeled. While these calls do not constitute quantitative validation, they are likely to reflect actual cell type assignments in cases when SMART-Seq v4 data is missing. Note that a value of \"Neuron\" indicates that the specific neuronal populations targeted could not be accurately assessed in that experiment. \n",
    "\n",
    "Note: check this for SSV4 - if this is missing then prediction is blank"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ff14da9f-887b-4326-9746-cac70e6e1dd7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MapMySectionsID</th>\n",
       "      <th>Genetics Tools Type</th>\n",
       "      <th>STPT Data File Path</th>\n",
       "      <th>STPT Thumbnail Image</th>\n",
       "      <th>Neuroglancer File Path</th>\n",
       "      <th>CCF Registered Image File Path</th>\n",
       "      <th>Target_Cell_Population</th>\n",
       "      <th>Qualitative Image Assessment</th>\n",
       "      <th>|</th>\n",
       "      <th>ABC.NN</th>\n",
       "      <th>...</th>\n",
       "      <th>Oligo.NN</th>\n",
       "      <th>Peri.NN</th>\n",
       "      <th>Pvalb.Gaba</th>\n",
       "      <th>Pvalb.chandelier.Gaba</th>\n",
       "      <th>SMC.NN</th>\n",
       "      <th>Sncg.Gaba</th>\n",
       "      <th>Sst.Chodl.Gaba</th>\n",
       "      <th>Sst.Gaba</th>\n",
       "      <th>VLMC.NN</th>\n",
       "      <th>Vip.Gaba</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MMS.training.001</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1285852977...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>Pax6</td>\n",
       "      <td>strong, sparse, GABAergic</td>\n",
       "      <td>|</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MMS.training.002</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1283915380...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>Pvalb</td>\n",
       "      <td>strong, dense, Astro</td>\n",
       "      <td>|</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MMS.training.003</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1121585094...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>Pvalb</td>\n",
       "      <td>strong, dense, GABAergic; strong, dense, L5_Gl...</td>\n",
       "      <td>|</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MMS.training.004</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1200969502...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>Endo</td>\n",
       "      <td>strong, sparse, Endo_Peri; weak, dense, Endo_P...</td>\n",
       "      <td>|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MMS.training.005</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1177798305...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>strong, dense, Oligo</td>\n",
       "      <td>|</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MMS.training.006</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1195060569...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>L4_IT</td>\n",
       "      <td>[not assessed], [not assessed], Neuron; strong...</td>\n",
       "      <td>|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MMS.training.007</td>\n",
       "      <td>Enhancer AAV</td>\n",
       "      <td>s3://allen-genetic-tools/tissuecyte/1171819575...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/allen-genet...</td>\n",
       "      <td>https://neuroglancer-demo.appspot.com/#!s3://a...</td>\n",
       "      <td>https://s3.us-west-2.amazonaws.com/map-my-sect...</td>\n",
       "      <td>L6_IT_Car3</td>\n",
       "      <td>strong, sparse, L6_Glutamatergic; weak, dense,...</td>\n",
       "      <td>|</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    MapMySectionsID Genetics Tools Type  \\\n",
       "0  MMS.training.001        Enhancer AAV   \n",
       "1  MMS.training.002        Enhancer AAV   \n",
       "2  MMS.training.003        Enhancer AAV   \n",
       "3  MMS.training.004        Enhancer AAV   \n",
       "4  MMS.training.005        Enhancer AAV   \n",
       "5  MMS.training.006        Enhancer AAV   \n",
       "6  MMS.training.007        Enhancer AAV   \n",
       "\n",
       "                                 STPT Data File Path  \\\n",
       "0  s3://allen-genetic-tools/tissuecyte/1285852977...   \n",
       "1  s3://allen-genetic-tools/tissuecyte/1283915380...   \n",
       "2  s3://allen-genetic-tools/tissuecyte/1121585094...   \n",
       "3  s3://allen-genetic-tools/tissuecyte/1200969502...   \n",
       "4  s3://allen-genetic-tools/tissuecyte/1177798305...   \n",
       "5  s3://allen-genetic-tools/tissuecyte/1195060569...   \n",
       "6  s3://allen-genetic-tools/tissuecyte/1171819575...   \n",
       "\n",
       "                                STPT Thumbnail Image  \\\n",
       "0  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "1  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "2  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "3  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "4  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "5  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "6  https://s3.us-west-2.amazonaws.com/allen-genet...   \n",
       "\n",
       "                              Neuroglancer File Path  \\\n",
       "0  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "1  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "2  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "3  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "4  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "5  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "6  https://neuroglancer-demo.appspot.com/#!s3://a...   \n",
       "\n",
       "                      CCF Registered Image File Path Target_Cell_Population  \\\n",
       "0  https://s3.us-west-2.amazonaws.com/map-my-sect...                   Pax6   \n",
       "1  https://s3.us-west-2.amazonaws.com/map-my-sect...                  Pvalb   \n",
       "2  https://s3.us-west-2.amazonaws.com/map-my-sect...                  Pvalb   \n",
       "3  https://s3.us-west-2.amazonaws.com/map-my-sect...                   Endo   \n",
       "4  https://s3.us-west-2.amazonaws.com/map-my-sect...                  Oligo   \n",
       "5  https://s3.us-west-2.amazonaws.com/map-my-sect...                  L4_IT   \n",
       "6  https://s3.us-west-2.amazonaws.com/map-my-sect...             L6_IT_Car3   \n",
       "\n",
       "                        Qualitative Image Assessment  |  ABC.NN  ...  \\\n",
       "0                          strong, sparse, GABAergic  |     NaN  ...   \n",
       "1                               strong, dense, Astro  |     NaN  ...   \n",
       "2  strong, dense, GABAergic; strong, dense, L5_Gl...  |     NaN  ...   \n",
       "3  strong, sparse, Endo_Peri; weak, dense, Endo_P...  |     0.0  ...   \n",
       "4                               strong, dense, Oligo  |     NaN  ...   \n",
       "5  [not assessed], [not assessed], Neuron; strong...  |     0.0  ...   \n",
       "6  strong, sparse, L6_Glutamatergic; weak, dense,...  |     0.0  ...   \n",
       "\n",
       "   Oligo.NN  Peri.NN  Pvalb.Gaba  Pvalb.chandelier.Gaba  SMC.NN  Sncg.Gaba  \\\n",
       "0       NaN      NaN         NaN                    NaN     NaN        NaN   \n",
       "1       NaN      NaN         NaN                    NaN     NaN        NaN   \n",
       "2       NaN      NaN         NaN                    NaN     NaN        NaN   \n",
       "3       0.0      0.0         3.0                    0.0     0.0        0.0   \n",
       "4       NaN      NaN         NaN                    NaN     NaN        NaN   \n",
       "5       0.0      0.0         0.0                    0.0     0.0        0.0   \n",
       "6       1.0      0.0         0.0                    0.0     0.0        0.0   \n",
       "\n",
       "   Sst.Chodl.Gaba  Sst.Gaba  VLMC.NN  Vip.Gaba  \n",
       "0             NaN       NaN      NaN       NaN  \n",
       "1             NaN       NaN      NaN       NaN  \n",
       "2             NaN       NaN      NaN       NaN  \n",
       "3             0.0       1.0      0.0       0.0  \n",
       "4             NaN       NaN      NaN       NaN  \n",
       "5             0.0       2.0      0.0       0.0  \n",
       "6             0.0       0.0      0.0       0.0  \n",
       "\n",
       "[7 rows x 36 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = workbook.parse(sheets[3])\n",
    "df.head(7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a89c2f9f-1590-455d-8e68-bcb8259e37e5",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.14"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
